M2/Macaulay2/packages/Msolve.m2

```diff
@@ -0,0 +1,588 @@
+newPackage(
+	"Msolve",
+	Version => "1.24.05", 
+    	Date => "July 2024",
+    	Authors => {{Name => "Martin Helmer", 
+		  Email => "mhelmer@ncsu.edu", 
+		  HomePage => "http://martin-helmer.com/"}, {Name => "Mike Stillman", 
+		  Email => "mike@math.cornell.edu", 
+		  HomePage => "https://math.cornell.edu/michael-e-stillman"},{Name => "Anton Leykin", 
+		  Email => "leykin@math.gatech.edu", 
+		  HomePage => "https://antonleykin.math.gatech.edu/"}},
+	  Keywords => {"Groebner Basis Algorithms" , "Interfaces"},
+    	Headline => "An interface to the msolve package (https://msolve.lip6.fr/) which computes Groebner Basis and does real root isolation",
+	PackageImports => { "Elimination", "Saturation" },
+    	AuxiliaryFiles => true,
+	DebuggingMode => false,       
+	OptionalComponentsPresent => run("command -v msolve > /dev/null") == 0
```

Comment:
I don't think `command -v` will be sufficient for this if msolve is built by us and installed in `currentLayout#"programs"`.  `findProgram` looks there, so that could potentially be used instead.  But of course, this line is run when calling `readPackage`, so we'd like to do as few things as possible.

Possible idea for another PR -- maybe `OptionalComponentsPresent` could be lazily evaluated when it's time to actually run tests or build examples?

Reply:
Actually, now that I think about it, we might be able to drop `OptionalComponentsPresent` completely if msolve is built and distributed with Macaulay2 if it isn't already present on the system.

Reply:
> Possible idea for another PR -- maybe `OptionalComponentsPresent` could be lazily evaluated when it's time to actually run tests or build examples?

#3369?

> Actually, now that I think about it, we might be able to drop `OptionalComponentsPresent` completely since msolve is built and distributed with Macaulay2 if it isn't already present on the system.

Yeah I guess there's no point.